src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -59488,6 +59520,2310 @@ class C {
+                Diagnostic(ErrorCode.ERR_PartialMethodInconsistentConstraints, "M1").WithArguments("A.M1<TF1>()").WithLocation(8, 14)
+                );
+
+            var source2 =
+@"
+partial class A
+{
+#nullable disable
+partial void M1<TF1>()
+{
+}
+
+#nullable enable
+partial void M1<TF1>();
+}
+";
+            var comp2 = CreateCompilation(source2);
+
+            comp2.VerifyDiagnostics(
+                // (5,14): error CS0761: Partial method declarations of 'A.M1<TF1>()' have inconsistent type parameter constraints
+                // partial void M1<TF1>()
+                Diagnostic(ErrorCode.ERR_PartialMethodInconsistentConstraints, "M1").WithArguments("A.M1<TF1>()").WithLocation(5, 14)
+                );
+        }
+
+        [Fact]
+        public void Constraints_132()
+        {
+            var source1 =
+@"
+partial class A
+{
+#nullable disable
+partial void M1<TF1>();
+
+partial void M1<TF1>() where TF1 : object
+{
+}
+}
+";
+            var comp1 = CreateCompilation(source1);
+
+            comp1.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void Constraints_133()
+        {
+            var source1 =
+@"
+partial class A
+{
+#nullable enable
+partial void M1<TF1>();
+
+partial void M1<TF1>() where TF1 : object
+{
+}
+}
+";
+            var comp1 = CreateCompilation(source1);
+
+            comp1.VerifyDiagnostics(
+                // (7,14): error CS0761: Partial method declarations of 'A.M1<TF1>()' have inconsistent type parameter constraints
+                // partial void M1<TF1>() where TF1 : object
+                Diagnostic(ErrorCode.ERR_PartialMethodInconsistentConstraints, "M1").WithArguments("A.M1<TF1>()").WithLocation(7, 14)
+                );
+        }
+
+        [Fact]
+        public void Constraints_134()
+        {
+            var source1 =
+@"
+partial class A
+{
+#nullable disable
+partial void M1<TF1>();
+
+#nullable enable
+partial void M1<TF1>() where TF1 : object
+{
+}
+}
+";
+            var comp1 = CreateCompilation(source1);
+
+            comp1.VerifyDiagnostics(
+                // (8,14): error CS0761: Partial method declarations of 'A.M1<TF1>()' have inconsistent type parameter constraints
+                // partial void M1<TF1>() where TF1 : object
+                Diagnostic(ErrorCode.ERR_PartialMethodInconsistentConstraints, "M1").WithArguments("A.M1<TF1>()").WithLocation(8, 14)
+                );
+        }
+
+        [Fact]
+        public void Constraints_135()
+        {
+            var source1 =
+@"
+partial class A
+{
+#nullable enable
+partial void M1<TF1>();
+
+#nullable disable
+partial void M1<TF1>() where TF1 : object
+{
+}
+}
+";
+            var comp1 = CreateCompilation(source1);
+
+            comp1.VerifyDiagnostics(
+                // (8,14): error CS0761: Partial method declarations of 'A.M1<TF1>()' have inconsistent type parameter constraints
+                // partial void M1<TF1>() where TF1 : object
```

Comment:
Because they differ by nullability only, this should be a nullable warning and suppressed in this disabled region.